src/DualQuaternion.jl

```diff
@@ -112,6 +109,7 @@ end
                                                               dq.q0 * dw.qe + dq.qe * dw.q0,
                                                               dq.norm && dw.norm)
 (/)(dq::DualQuaternion, dw::DualQuaternion) = dq * inv(dw)
+(==)(q::DualQuaternion, w::DualQuaternion) = (q.q0 == w.q0) & (q.qe == w.qe) # ignore .norm field
```

Comment:
Ah, I see you mentioned `&` is better in https://github.com/JuliaGeometry/Quaternions.jl/pull/52#issuecomment-1003257187. Can you explain the reasoning?

Reply:
The `code_native` for the `Quaternion` `&` version is great. It compare all four fields via a single SIMD instruction and has no branches. Meanwhile, the same function using `&&` requires a check-then-branch for each field because it must short-circuit the later comparisons on inequality.

The benchmarking isn't that different between the two. About 1.9ns for `&` and 2.1ns for `&&`. The benchmark I used would have let the branch predictor learn the `&&` version, so `&&` might perform worse in the wild. Also, `&` is easier to inline since it's shorter.

The same story holds for `DualQuaternion`. I didn't check `Octonion` but the story should be similar to `DualQuaternion` or slightly better (since the SIMD loads don't have to hopscotch over the `norm` field of the first Quaternion.)

Reply:
Thanks!